File: code/cpp_test/presentation_code/presentation.py
import torch

model = torch.nn.Linear(num_features, num_classes)
optimizer = torch.optim.SGD(model.parameters(), lr=0.1)
data_loader = torch.utils.data.DataLoader(dataset)

for epoch in range(num_epochs):
    for input_data, label in data_loader:
        prediction = model.forward(input_data)
        loss = loss_function(prediction, label)
        loss.backward()
        optimizer.step()

File: code/cpp_test/presentation_code/presentation.cpp
#include <torch/torch.h>

torch::nn::Linear model(num_features, num_classes);
torch::optim::SGD optimizer(model.parameters(), torch::optim::SGDOptions(0.1));
torch::data::DataLoader data_loader(dataset);

for(size_t epoch = 0; epoch < num_epochs; ++epoch) 
{
    for(auto [input_data, label] : data_loader) 
    {
        auto prediction = model->forward(data);
        auto loss = loss_function(prediction, label);
        // auto loss = torch::nn::functional::cross_entropy(output, target);
        loss.backward();
        optimizer.step();
    }
}